src/Symfony/Component/Messenger/MessageRecorderInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+use Symfony\Contracts\Service\ResetInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Matthias Noback <matthiasnoback@gmail.com>
+ */
+interface MessageRecorderInterface extends ResetInterface
+{
+    /**
+     * Fetch recorded messages.
+     *
+     * @return object[]
+     */
+    public function fetch(): array;
```

Comment:
This name seems weird to me. Like I said previously my model will implement this interface. For example, a product model. My product will have Product::fetch() and Product::reset(). In term of business that does not have any sense. I think we should use `eraseMessage()` and `recordedMessage()` it more explicit. I think it is a good compromise between technical and business stuff.